project/SparkBuild.scala

```diff
@@ -65,10 +65,10 @@ object BuildCommons {
   ).map(ProjectRef(buildLocation, _)) ++ sqlProjects ++ streamingProjects ++ connectProjects
 
   val optionallyEnabledProjects@Seq(kubernetes, yarn,
-    sparkGangliaLgpl, streamingKinesisAsl,
+    sparkGangliaLgpl, streamingKinesisAsl, profiler,
```

Comment:
After enabling `-Pjvm-profiler`, the `sbt-pom-reader` plugin will complete the import of the SBT configuration for the `profiler` module. Therefore, if we only wish to trigger the compilation of this module during the GitHub Action execution, there is no need to modify `SparkBuild.scala`.

Reply:
@LuciferYang thanks for the guidance, after some local experiments, I think we must modify `SparkBuild.scala` to explicitly list the `profiler` module, otherwise `MimaKeys.mimaFailOnNoPrevious := false` would take effect.